lib/hook.js

```diff
@@ -57,40 +73,37 @@ Hook.HOOK_PRIORITY_DEFAULT = 1000;
  * Scans the specified path for hooks. Each hook is identified by a JavaScript
  * file.
  * @param {String} dir - A directory or JavaScript file to be scanned.
+ * @param {RegExp} include - Regular expression to match for included filenames.
+ * @param {RegExp} exclude - Regular expression to match for excluded filenames.
  */
-Hook.prototype.scanHooks = function scanHooks(dir) {
+Hook.prototype.scanHooks = function scanHooks(dir, include, exclude) {
 	if (!this.hooks.scannedPaths[dir] && fs.existsSync(dir)) {
-		var jsfile = /\.js$/,
-			ignore = /^[._]/,
-			isDir = fs.statSync(dir).isDirectory();
+		const fileList = findInDir(dir, include, exclude);
 
-		(isDir ? fs.readdirSync(dir) : [ dir ]).forEach(function (filename) {
-			var file = isDir ? path.join(dir, filename) : filename;
-			if (fs.existsSync(file) && fs.statSync(file).isFile() && jsfile.test(filename) && (!isDir || !ignore.test(path.basename(file)))) {
-				try {
-					vm.runInThisContext('(function (exports, require, module, __filename, __dirname) { ' + fs.readFileSync(file).toString() + '\n});', file, 0, false);
-					var mod = require(file);
-					if (mod.id) {
-						Array.isArray(this.hooks.ids[mod.id]) || (this.hooks.ids[mod.id] = []);
-						this.hooks.ids[mod.id].push({
-							file: file,
-							version: mod.version || null
-						});
-						// don't load duplicate ids
-						if (this.hooks.ids[mod.id].length > 1) {
-							return;
-						}
-					}
-					if (!this.version || !mod.cliVersion || semver.satisfies(appc.version.format(this.version, 0, 3, true), mod.cliVersion)) {
-						mod.init && mod.init(this.logger, this.config, this, appc);
-						this.hooks.loadedFilenames.push(file);
-					} else {
-						this.hooks.incompatibleFilenames.push(file);
+		fileList.forEach(function (file) {
```

Comment:
Same as above. for-of is amazing.

Reply:
I was trying to limit the changes to just this feature.  `forEach` is used throughout this file including here.  It is always used in the context of passing `this` as the additional parameter.  I want to make sure we are not breaking anything by going to `for...of` without `this`